### ISSUE TITLE
bulk import ?

### ISSUE THREAD

**Issue by txsastre** [ssaring/sportstracker#189]
hi there just downloaded this app, seems great.

I wanted to go out of endomondo and garmin, and I have a lot of exercises.

it there a way to import all of them at once to ST ?

thank you !

**Comment by ssaring** [comment]
Currently not, I think there was a similar feature request (but not in work yet).

I would suggest you to inspect the XML format of the exercise file named 'exercises.xml'. It's very simple and structured - it should be quite easy to write a script which converts your existing exercise data and add it to this file. Most of the online platforms has an export feature, but I don't tested it yet.

Good luck!

**Comment by txsastre** [comment]
yes I read that, it was years ago

yes I have knowledge to do that, I've seen the structure of the file, and does not seems so difficult.
just one question, how can I read the information from a .FIT file ? (garmin Forerunner 620)
If I do a "cat" or "type" seems that's in binary form.

thanks

**Comment by txsastre** [comment]
the idea is to put a watch watching the usb, if the garmin is connected then copy the new files to my ssd (which folder is on a nextcloud). then read the FIT file to retrieve the summarize and then add it to the 'exercises.xml' also the path where the FIT will be stored.

as I see that your app reads FIT files I would like how, so I could read it for the summarize info (and put in the 'exercises.xml)

Thanks again.

**Comment by ssaring** [comment]
Yes, FIT is a binary format. But it's very good documented, there's an API + documentation too.
Just take a look to my class GarminFitParser, it should be helpful for for your import too.

**Comment by txsastre** [comment]
ok. thank you. 
I will keep an eye on it (if I'm lucky and I have enough time for that)

Thanks again for the help and thanks for the app !

**Comment by mrkobel** [comment]
Hi, I came here with exactly the same question. I spent some time looking for an app like yours and it seems like exaclty what I need. I want something offline, independent from online platforms for privacy reasons. I have few years of tracks / activities from Endomondo and Strava, but seems like there is no way to do bulk import. Unfortunately I'm very rusty with my coding skills and can't do this myself... Is there any way I can vote for this feature request? :)

cheers!

**Comment by ssaring** [comment]
I'm not aware that anybody is currently working on that. A bulk import is also very specific for the particular source platform, that's why I don't think there will be a generic solution which covers all of them.

**Comment by nusibrains** [comment]
Same request  here.

I can import one FIT file by using 'the import from hrm' feature, it works very well.
I don't got skills to push it, but I think it should be something like a loop of the current 'import from hrm' feature
